Hackathon winners creativity (#38)
dot-layer/blog: content/blog/2018-11-17-hackathon-winners-creativity.md

#### Q: Selon vous, pensez-vous que c'est aussi l'élément déterminant qui a fait pencher la balance chez les juges?

Définitivement!

#### Nous avons posé la même question aux juges de cette catégorie. Rappelons-nous que les critères d'évaluation pour cette catégorie étaient à leur discrétion. Voici leur réponse:

L'inclusion de techniques de photographie à la solution était effectivement l'élément clé dans notre décision. Les organisateurs avaient fourni des pistes de solution, mais *Les Beans* ont osé emprunter un chemin différent. Ça a vraiment démontré que l'équipe était capable de *think outside the box*. Ça démontrait aussi une curiosité au sujet de la problématique à résoudre.

#### Q: Pouvez-vous nous partager les meilleurs extraits de votre solution?

Voici quelques exemples qui illustrent l'effet des filtres sur les images.

| Avant          | Après            |
:---------------:|:-----------------:
![](image-2.png) | ![](image-2.jpg)
![](image-4.png) | ![](image-4.jpg)
![](image-7.png) | ![](image-7.jpg)
![](image-13.png) | ![](image-13.jpg)

> 

Reviewer: @philbl, tu penses que tu peux commenter de quoi ici? Scuse Laurent je suis en retard un peu.

Reviewer 2: @samperochkin Des choses à rajouter ?

Définitivement!

#### Nous avons posé la même question aux juges de cette catégorie. Rappelons-nous que les critères d'évaluation pour cette catégorie étaient à leur discrétion. Voici leur réponse:

L'inclusion de techniques de photographie à la solution était effectivement l'élément clé dans notre décision. Les organisateurs avaient fourni des pistes de solution, mais *Les Beans* ont osé emprunter un chemin différent. Ça a vraiment démontré que l'équipe était capable de *think outside the box*. Ça démontrait aussi une curiosité au sujet de la problématique à résoudre.

#### Q: Pouvez-vous nous partager les meilleurs extraits de votre solution?

Voici quelques exemples qui illustrent l'effet des filtres sur les images.

| Avant          | Après            |
:---------------:|:-----------------:
![](image-2.png) | ![](image-2.jpg)
![](image-4.png) | ![](image-4.jpg)
![](image-7.png) | ![](image-7.jpg)
![](image-13.png) | ![](image-13.jpg)

#### Q: Que conseillez-vous aux gens qui veulent commencer à participer à des hackathons?

**JT**: Simplement avoir de l'ouverture d'esprit, être créatif et vouloir en apprendre plus! Tout le monde sort gagnant de ces événements.

> **SP**: De ne pas se laisser intimider par l'aspect compétitif. Il y avait des gens de tous les niveaux et l'ambiance en était vraiment au partage de connaissances.

Reviewer: ```suggestion
**SP**: De ne pas se laisser intimider par l'aspect compétitif. Il y avait des gens de tous les niveaux et l'ambiance était à la collaboration.
```

#### Q: Quel est le background des membres de l'équipe? Croyez-vous que ce background vous favorisait dans la catégorie Leonardo da Vinci?

**JT**: Domaine de l'actuariat, consultation internationale. Je crois que la créativité vient plus de mes intérêts personnels et de mon ouverture d'esprit. 

**SP**: Les trois avons étudié, à un certain moment, en actuariat. Maintenant, on se concentre plus particulièrement en statistique et apprentissage machine, que ce soit dans d'autres facultés ou de manière autodidacte. Par rapport aux autres, ça ne nous a probablement pas aidé plus qu'il le faut en ce qui concerne la créativité. Pour être honnête, on visait directement cette catégorie en commençant. Et le fait que les autres équipes les plus créatives ont gagné les prix de la présentation et de la qualité du code nous a probablement aidé un peu!

#### Q: De quelle façon votre équipe a-t-elle alloué les 6 heures à sa disposition? Avec du recul, les auriez-vous allouées de façon différente?

> **SP**: Après environ une heure, on avait déjà du code qui faisait l'essentiel du travail : un modèle linéaire généralisé, un réseau de neurones et une forêt aléatoire (*Random Forest*). Nos trois modèles étaient assez similaires en termes de performances. 

Reviewer: ```suggestion
**SP**: Après environ une heure, on avait déjà du code qui faisait l'essentiel du travail : un modèle linéaire généralisé, un réseau de neurones et un ensemble d'arbres de décisions. Merci au librairies R keras, glmnet et xgboost! On avait encore de petits bugs avec nos données par contre.
```

> Je crois que j'ai passé trop de temps à faire de la recherche en grille (*grid search*) pour trouver de meilleurs hyperparamètres. En rétrospective, j'aurais aimé passez plus de temps pour inclure dans la présentation certains aspects subtiles de nos modèles. Par exemple, on a utilisé des poids variables pour nos observations provenant de différentes classes, question de favoriser la détection des toits verts quitte à ajouter des faux positifs dans nos prédictions. J'ai l'impression que personne n'a fait ça aussi et ça aurait été intéressant de le partager. On a peut-être commencé à préparer notre présentation trop 

Reviewer: ```suggestion
Nous avons passé trop de temps à faire de la recherche en grille (*grid search*) pour trouver de meilleurs hyperparamètres. En rétrospective, j'aurais aimé passez plus de temps pour inclure dans la présentation certains aspects subtiles de nos modèles. On a aussi commencé à préparer notre présentation trop 
```
